core/src/main/java/com/datastrato/gravitino/meta/MetalakeUser.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.meta;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.Field;
+import com.datastrato.gravitino.HasIdentifier;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.User;
+import com.google.common.collect.Maps;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import lombok.ToString;
+
+/** A class representing a user metadata entity in Gravitino. */
+@ToString
+public class MetalakeUser implements User, Entity, Auditable, HasIdentifier {
```

Comment:
Not to choose to name this `CombinedUser` Because the underlying system may have different users. We can't map them one to one.
Not to choose to name this `BaseUser`. Because BaseUser looks like an abstract class although we already have a class `BaseMetalake`.

Reply:
I don't like this name, user is just a user, `MetalakeUser` will make me think that this user is only for metalake.

Reply:
How about `concreteUser`, `UserImpl` and `BasicUser`?

Reply:
How about ManagedUser?

Reply:
How about ManagedUser?